src/service/chat.js

```diff
@@ -161,11 +161,55 @@
   }
 };
 
+const moveChatToSigned = async ({oldChannelId, targetUserId, source}) => {
+  try {
+    const response = await api.post('/chat/move-to-sign', {oldChannelId, targetUserId, source});
+    if (response.status === 200) {
+      return Promise.resolve(response.data);
+    }
+    return Promise.reject(response.data);
+  } catch (e) {
+    if (e?.response?.data?.message) return Promise.reject(e?.response?.data?.message);
+    return Promise.reject(e);
+  }
+};
+
+const moveChatToAnon = async ({
+  anon_user_info_color_code,
+  anon_user_info_color_name,
+  anon_user_info_emoji_code,
+  anon_user_info_emoji_name,
+  oldChannelId,
+  targetUserId,
+  source
+}) => {
+  try {
+    const response = await anonymousApi.post('/chat/move-to-anon', {
+      anon_user_info_color_code,
+      anon_user_info_color_name,
+      anon_user_info_emoji_code,
+      anon_user_info_emoji_name,
+      oldChannelId,
+      targetUserId,
+      source
+    });
+    if (response.status === 200) {
+      return Promise.resolve(response.data);
+    }
+    return Promise.reject(response.data);
+  } catch (e) {
+    if (e?.response?.data?.message) return Promise.reject(e?.response?.data?.message);
+    return Promise.reject(e);
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Similar to the previous comment, the error handling in the `moveChatToAnon` function could also be improved by adding a generic error message as a fallback.
```diff
-    if (e?.response?.data?.message) return Promise.reject(e?.response?.data?.message);
-    return Promise.reject(e);
+    const errorMessage = e?.response?.data?.message || 'An error occurred while moving chat to anonymous';
+    return Promise.reject(errorMessage);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->